src/devices/Ft232H/Ft232HGpio.cs

```diff
@@ -0,0 +1,197 @@
+            }
+
+            DeviceInformation._PinOpen[pinNumber] = true;
+        }
+
+        /// <inheritdoc/>
+        protected override void ClosePin(int pinNumber)
+        {
+            DeviceInformation._PinOpen[pinNumber] = false;
+        }
+
+        /// <inheritdoc/>
+        protected override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            if (pinNumber < 8)
+            {
+                if (mode != PinMode.Output)
+                {
+                    byte mask = 0xFF;
+                    mask &= (byte)(~(1 << pinNumber));
+                    DeviceInformation._gpioLowDir &= mask;
+                }
+                else
+                {
+                    DeviceInformation._gpioLowDir |= (byte)(1 << pinNumber);
+                }
+
+                DeviceInformation.SetGpioValuesLow();
+            }
+            else
+            {
+                if (mode != PinMode.Output)
+                {
+                    byte mask = 0xFF;
+                    mask &= (byte)(~(1 << (pinNumber - 8)));
+                    DeviceInformation._gpioHighDir &= mask;
+                }
+                else
+                {
+                    DeviceInformation._gpioHighDir |= (byte)(1 << (pinNumber - 8));
+                }
+
+                DeviceInformation.SetGpioValuesHigh();
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override PinMode GetPinMode(int pinNumber)
+        {
+            if (pinNumber < 8)
+            {
+                return ((DeviceInformation._gpioLowDir >> pinNumber) & 0x01) == 0x01 ? PinMode.Output : PinMode.Input;
+            }
+
+            return ((DeviceInformation._gpioHighDir >> (pinNumber - 8)) & 0x01) == 0x01 ? PinMode.Output : PinMode.Input;
+        }
+
+        /// <inheritdoc/>
+        protected override bool IsPinModeSupported(int pinNumber, PinMode mode)
+        {
+            if ((mode == PinMode.InputPullDown) || (mode == PinMode.InputPullUp))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <inheritdoc/>
+        protected override PinValue Read(int pinNumber)
+        {
+            if (pinNumber < 8)
+            {
+                var val = DeviceInformation.GetGpioValuesLow();
+                return (((val >> pinNumber) & 0x01) == 0x01) ? PinValue.High : PinValue.Low;
+            }
+            else
+            {
+                var valhigh = DeviceInformation.GetGpioValuesHigh();
+                return (((valhigh >> (pinNumber - 8)) & 0x01) == 0x01) ? PinValue.High : PinValue.Low;
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void Write(int pinNumber, PinValue value)
+        {
+            if (pinNumber < 8)
+            {
+                if (value == PinValue.High)
+                {
+                    DeviceInformation._gpioLowData |= (byte)(1 << pinNumber);
```

Comment:
would it make sense to have a single uint16 for both low and high and make SetGpioValues() write both? I think that should be fairly cheap

Reply:
treat as optional since this is an implementation detail although I'd prefer we didn't touch fields starting with underscore unless in the same class